util/fp-finder/spell.sh

```diff
@@ -1,20 +1,82 @@
 #!/bin/bash
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+if ! hash spell > /dev/null 2>&1; then
+    echo "This program requires spell to be installed. Aborting"
+    exit 1
+fi
 
-RULES_DIR="${SCRIPT_DIR}/../../rules/"
+check() {
+    local datafile="${1}"
+    local datafile_name="${datafile##*/}"
 
-for datafile in "$RULES_DIR"*.data; do
-    DATAFILE_NAME=${datafile##*/}
-    
-    echo "-> checking ${DATAFILE_NAME}"
+    if ! ${PARSEABLE}; then
+        echo "-> checking ${DATAFILE_NAME}"
+    fi
 
     for word in $(grep -E '^[a-z]+$' "${datafile}" | uniq | sort); do
         IS_NOT_ENGLISH=$(echo "${word}" | spell | wc -l)
         if [ "${IS_NOT_ENGLISH}" -lt 1 ]; then
-            echo "   \`- found English word: ${word}"
+        if true; then
+            if ! ${PARSEABLE}; then
+                printf "   \`- found English word: "
+            fi
         fi
+        echo "${word}"
     done
 
-    echo ""
+    if ! ${PARSEABLE}; then
+        echo ""
+    fi
+}
+
+usage() {
+    cat <<EOF
+usage: spell.sh [-ph] [file]
+    Finds English words in files that contain word lists.
+
+    The optional file argument is the path to a file you want to check. If omitted,
+    all files with the .data suffix in the rules directory will be searched.
+
+    -h, --help     \tShow this message and exit
+    -p, --parseable\tPrint machine readable output
```

Comment:
The problem is when parsing, or when doing the output? I think it isi a bit confusing...